setup.cfg

```diff
@@ -43,14 +43,14 @@ console_scripts =
 [options.extras_require]
 doc =
     furo
-    sphinx
-    sphinx-autodoc-typehints
-    sphinxcontrib-autoprogram
+    sphinx~=3.0
```

Comment:
I guess the same here too and the following, any reason we need compatibility and not just take whatever is latest?

Reply:
I find that is harder to detect issues in sphinx as it sometimes just ends up wrongly formatting the docs, without raising an error or warning. I'd prefer updating to new major sphinx versions manually, but if you have a strong opinion on this I can change.